lib/resources/resource.go

```diff
@@ -0,0 +1,152 @@
+	OSName      string `yaml:"osName"`
+	OSVersion   string `yaml:"osVersion"`
+	Tags        string `yaml:"tags"`
+	Username    string `yaml:"username"`
+	SSHKeypath  string `yaml:"ssh-keypath"`
+}
+
+// Result struct holds Zabbix hosts info
+type Result []struct {
+	HostID      string `json:"hostid"`
+	Host        string `json:"host"`
+	Name        string `json:"name"`
+	Description string `json:"description"`
+	Triggers    []struct {
+		Triggerid   string `json:"triggerid,omitempty"`
+		Description string `json:"description,omitempty"`
+		Status      string `json:"status"`
+	} `json:"triggers"`
+}
+
+func (z *Zabbix) getInfo(apiKey, URL string) (Result, error) {
+	api := z.API
+	api.Key = apiKey
+	api.URL = URL
+
+	params := struct {
+		Output         []string `json:"output"`
+		SelectTriggers []string `json:"selectTriggers"`
+	}{
+		Output:         []string{"host", "name", "description"},
+		SelectTriggers: []string{"description", "status"},
+	}
+
+	payload := api.BuildPayload(params, "host.get")
+
+	resp, err := api.MakeRequest(payload)
+	if err != nil {
+		return nil, fmt.Errorf("cannot make API request. error: %v", err)
+	}
+
+	var r struct {
+		Result `json:"result"`
+	}
+
+	err = json.NewDecoder(resp.Body).Decode(&r)
+	if err != nil {
+		return nil, fmt.Errorf("cannot decode response. error: %v", err)
+	}
+
+	return r.Result, nil
+}
+
+// MakeResource generates Rundeck resource model document in Yaml format
+// with hosts' information from Zabbix
+func (z *Zabbix) MakeResource(apiKey, URL, file string) error {
+	info, err := z.getInfo(apiKey, URL)
+	if err != nil {
+		return err
+	}
+
+	var r Resource
+
+	for _, v := range info {
+		// convert item to json
+		jsonString, err := json.Marshal(v)
+		if err != nil {
+			return fmt.Errorf("cannot convert to json. error: %v", err)
+		}
+
+		d, err := r.toYaml(jsonString)
+		if err != nil {
+			return err
+		}
+
+		if err := dumpToFile(file, d); err != nil {
+			return err
+		}
+	}
+
+	return nil
+}
+
+/*
+toYaml converts given json string to output similar to:
+
+dummy-host:
+  hostname: dummy-host
+  nodename: dummy-host
+  description: ""
+  osArch: ""
+  osFamily: ""
+  osName: ""
+  osVersion: ""
+  tags: ""
+  username: ""
+  ssh-keypath: ""
+*/
+func (r *Resource) toYaml(jsonString []byte) ([]byte, error) {
+	// convert json string to struct
+	json.Unmarshal(jsonString, &r)
+
+	m := make(map[string]Resource)
+	m[r.Name] = *r
+	d, err := yaml.Marshal(&m)
+	if err != nil {
+		return nil, fmt.Errorf("cannot unmarshal resource. error: %v", err)
+	}
+
+	return d, nil
+}
+
+func dumpToFile(filePath string, data []byte) error {
+	// If the filePath doesn't exist, create it, or append to the file
+	f, err := os.OpenFile(filePath, os.O_APPEND|os.O_CREATE|os.O_WRONLY, 0644)
+	if err != nil {
+		return fmt.Errorf("cannot open file. error: %v", err)
+	}
+
+	if _, err := f.Write(data); err != nil {
+		return fmt.Errorf("cannot write to file. error: %v", err)
+	}
+
+	if err := f.Close(); err != nil {
```

Comment:
There is little chance for this to error for ordinary files in the system. 
You could well defer file closure.